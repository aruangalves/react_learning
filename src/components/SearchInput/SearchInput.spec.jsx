import { render, screen } from "@testing-library/react";
import { SearchInput } from ".";
import userEvent from "@testing-library/user-event";

const placeholder = 'Busca por t√≠tulo';

describe('<SearchInput />', ()=>{
    it('should have a search value', ()=>{
        const fn = jest.fn();
        render(<SearchInput searchHandler={fn} searchValue={'testing'} />);

        const input = screen.getByPlaceholderText(placeholder);

        expect(input).toBeInTheDocument();
        expect(input.value).toBe('testing');

        
    });

    it('should call handleChange function on each key press', ()=>{
        const fn = jest.fn();
        render(<SearchInput searchHandler={fn} />);

        const input = screen.getByPlaceholderText(placeholder);

        const value = 'some search value';

        userEvent.type(input, value);
        expect(input.value).toBe(value);
        expect(fn).toHaveBeenCalledTimes(value.length);

        
    });

    it('should match snapshot', ()=>{
        const fn = jest.fn();
        const {container} = render(<SearchInput searchHandler={fn} />);
        const {firstChild} = container;

        expect(firstChild).toMatchSnapshot();
    });
});