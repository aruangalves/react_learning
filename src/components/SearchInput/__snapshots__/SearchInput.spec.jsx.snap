// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match snapshot 1`] = `
<input
  class="searchInput"
  placeholder="Busca por tÃ­tulo"
  type="search"
  value=""
/>
`;
